File: module_video/src/main/java/com/yyxnb/module_video/db/VideoDao.java
package com.yyxnb.module_video.db;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Query;

import com.yyxnb.network.db.BaseDao;
import com.yyxnb.module_video.bean.TikTokBean;

import java.util.List;

@Dao
public interface VideoDao extends BaseDao<TikTokBean> {

    @Query("SELECT * FROM video limit 0,20")
    LiveData<List<TikTokBean>> getVideos();

    //删全部
    @Query("DELETE FROM video")
    void deleteAll();

}

File: module_video/src/main/java/com/yyxnb/module_video/db/VideoDao.kt
package com.yyxnb.module_video.db

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Query
import com.yyxnb.common_base.db.BaseDao
import com.yyxnb.module_video.bean.TikTokBean

@Dao
interface VideoDao : BaseDao<TikTokBean> {

    @get:Query("SELECT * FROM video limit 0,20")
    val videos: LiveData<List<TikTokBean>>

    //删全部
    @Query("DELETE FROM video")
    fun deleteAll()
}